### PR TITLE
Document AStar

### PR DESCRIPTION
**Pull request by ezefran**

### CODE REVIEW

**Comment by mrawlingst** [comment]
I really recommend adding an example for each function. It'd help to see the example in usage.

See #11023 for example.

File: doc/base/classes.xml [hunk 1]
```diff
@@ -2556,7 +2558,7 @@
 			<argument index="2" name="weight_scale" type="float" default="1.0">
 			</argument>
 			<description>
-				Add a new point at the given position [code]pos[/code] with the given identifier [code]id[/code]. The [code]weight_scale[/code] has to be 1 or larger.
+				Adds a new point at the given position with the given identifier. A point with a lower [code]weight_scale[/code] will be prefered to form a path. The [code]id[/code] must be positive and the [code]weight_scale[/code] must be 1 or larger.
```
Comment (NathanLovato) [inline doc/base/classes.xml]: Avoid the passive voice: `will be prefered` -> `Prefers points with a lower weight_scale` or `Prioritizes points with...`.
The function is abstract. Could you add a code example maybe?

File: doc/base/classes.xml [hunk 2]
```diff
@@ -2613,7 +2616,7 @@
 			<argument index="0" name="to_pos" type="Vector3">
 			</argument>
 			<description>
-				Returns the id of closest point of given point.  -1 is returned if there are no points on AStar.
+				Returns the id of the closest point to the given position. -1 is returned if there are no points in the points pool.
```
Comment (NathanLovato) [inline doc/base/classes.xml]: Suggestion without passive voice: `Returns the closest point's id. Returns -1 if there are no points in the points pool.`

**Review by NathanLovato — COMMENTED** [review]
Avoid passive voice: https://github.com/GDquest/godot-3-guides/blob/master/contribution-guides/class-reference-styleguide.md

**Comment by akien-mga** [comment]
Could you squash the three commits together? See http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase

**Pull request merged**